compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -116,8 +117,19 @@ pub fn report_unstable(
         if is_soft {
             soft_handler(SOFT_UNSTABLE, span, &msg)
         } else {
-            feature_err_issue(&sess.parse_sess, feature, span, GateIssue::Library(issue), &msg)
-                .emit();
+            let mut err =
+                feature_err_issue(&sess.parse_sess, feature, span, GateIssue::Library(issue), &msg);
+            if using_vec_allocator_api {
+                if let Ok(snippet) =
+                    sm.span_to_snippet(sm.span_extend_to_prev_char(span, '<', true))
+                {
+                    err.help(&format!(
```

Comment:
We should emit a suggestion here rather than a help. See [`Diagnostic::span_suggestion`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_errors/diagnostic/struct.Diagnostic.html#method.span_suggestion).